### PR TITLE
Add notes on testing to developer doc

### PR DESCRIPTION
**Pull request by bgeihe**

### CODE REVIEW

File: docs/src/developers.md [hunk 1]
```diff
@@ -39,4 +39,46 @@ To create a new libtrixi release, follow these steps:
...
+
+### Testing the C-interface
+
+For testing the C-interface part of libtrixi we rely on [GoogleTest](https://google.github.io/googletest).
+The tests are contained in `cpp`-files located under `test/c`. They are processed by `cmake` and made available via
+`ctest`, provided the options
+```
+-DENABLE_TESTING=ON -DJULIA_PROJECT_PATH=<libtrixi-julia_directory>
+```
+were given. The executables can then be found under `<build_directory>/test/c` (they will not be installed). To run them execute
```
Comment (sloede) [inline docs/src/developers.md:55]:
```suggestion
are passed to `cmake` during configuration
The executables can then be found under `<build_directory>/test/c` (they will not be installed). To run them, execute
```

File: docs/src/developers.md [hunk 2]
```diff
@@ -39,4 +39,46 @@ To create a new libtrixi release, follow these steps:
...
+```
+ctest -V -R <regex>
+```
+from `<build_directory>/test/c` or the top-level directory `<build_directory>`.
+The optional argument `-V` turns on verbose output, and `-R` lets you specify a regular expression to select specific tests.
+A list of available tests can be obtained via `ctest -N`
+
+### Testing the Fortran-interface
+
+For testing the Fortran-interface part of libtrixi we rely on [test-drive](https://github.com/fortran-lang/test-drive),
```
Comment (sloede) [inline docs/src/developers.md:65]:
```suggestion
### Testing the Fortran interface

For testing the Fortran interface of libtrixi we rely on [test-drive](https://github.com/fortran-lang/test-drive),
```
As above?

File: docs/src/developers.md [hunk 3]
```diff
@@ -39,4 +39,46 @@ To create a new libtrixi release, follow these steps:
...
+which integrates with `cmake` and `ctest` as well. The tests are contained in `f90`-files under `test/fortran`. Usage is
+analogous to the C-interface part.
```
Comment (sloede) [inline docs/src/developers.md:67]:
```suggestion
analogous to the C interface part.
```

File: docs/src/developers.md [hunk 4]
```diff
@@ -39,4 +39,46 @@ To create a new libtrixi release, follow these steps:
...
+
+### Testing the Julia part
```
Comment (sloede) [inline docs/src/developers.md:69]:
```suggestion
### Testing the Julia interface
```
for consistency?

File: docs/src/developers.md [hunk 5]
```diff
@@ -39,4 +39,46 @@ To create a new libtrixi release, follow these steps:
...
+
+For testing the Julia part `LibTrixi.jl` of libtrixi we rely on
```
Comment (sloede) [inline docs/src/developers.md:71]:
```suggestion
For testing the Julia interface of libtrixi, which is contained in the Julia package `LibTrixi.jl`, we rely on
```

File: docs/src/developers.md [hunk 6]
```diff
@@ -39,4 +39,46 @@ To create a new libtrixi release, follow these steps:
...
+[Julia's testing infrastructure](https://docs.julialang.org/en/v1/stdlib/Test/), i.e. there is a dedicated test project,
+located under `LibTrixi.jl/test`, which contains `runtest.jl` and further `jl`-files containing the actual tests. Invoke
+julia via
```
Comment (sloede) [inline docs/src/developers.md:74]:
```suggestion
[Julia's testing infrastructure](https://docs.julialang.org/en/v1/stdlib/Test/). There is a dedicated test project,
located under `LibTrixi.jl/test`, which contains `runtest.jl` and further `jl`-files containing the actual tests. Invoke
Julia via
```

File: docs/src/developers.md [hunk 7]
```diff
@@ -39,4 +39,46 @@ To create a new libtrixi release, follow these steps:
...
+```
+JULIA_DEPOT_PATH=<julia-depot_directory> \
+LIBTRIXI_DEBUG=all \
+    julia --project=./LibTrixi.jl
+```
+and use
+```
+julia> import Pkg
+julia> Pkg.test()
+```
```
Comment (sloede) [inline docs/src/developers.md:84]:
~~~suggestion
```
JULIA_DEPOT_PATH=<julia-depot_directory> \
LIBTRIXI_DEBUG=all \
    julia --project=./LibTrixi.jl -e 'import Pkg; Pkg.test()'
```
~~~
Combine into a single command for easier copy pasting?

**Comment by bgeihe** [comment]
Many thanks for the hints!

**Review by sloede — APPROVED** [review]
LGTM!

**Pull request merged**